File: programme/digital-root/DigitalRoot.java
// Converts a number to its digital root
// The digital root of a number is defined as the sum of each digit defining the original number
// example 127 -> 1 + 2 + 7 -> 10
//          10 -> 1 + 0 -> 1

import java.util.Scanner;

public class DigitalRoot {
    public static void convert(int a) {
        // Ramans formula for calcuating the digital root
        System.out.println(1 + (a - 1) % 9);
    }

    public static void main(String[] args) {
        int a = 0;
        Scanner sc = new Scanner(System.in);
        System.out.print("Input  : ");
        a = sc.nextInt();
        System.out.print("\nThe digital root is : ");
        convert(a);
        sc.close();
    }
}

File: programme/digital-root/digital_root.go
// digital root is the sum of the digits of a number.
package main

import "fmt"

func main() {
	var n int
	fmt.Print("Enter the number : ")
	fmt.Scan(&n)
	fmt.Println("The digital root of the number is :", digitalRoot(n))
}

func digitalRoot(n int) int {
	for n > 9 {
		n = sumOfDigits(n)
	}
	return n
}

func sumOfDigits(n int) int {
	var sum int
	for n > 0 {
		sum += n % 10
		n /= 10
	}
	return sum
}
